### DISCUSSION TITLE
unrecoverable collision

### DISCUSSION THREAD

**Discussion by jeefm** [tagyoureit/nodejs-poolController#595] [Nixie]
Tossed Inbound Bytes 101,61,50,13,10,91,48,45,48,58,52,56,58,48,93,60,117,97,114,116,62,32,114,101,99,118,58,49,48,45,98 due to an unrecoverable collision.

What does it mean?

**Comment by rstrouse** [comment]
RS485 can be lossy because any tardiness from any responder can cause a collision. This is normal and njsPC is simply clearing out its buffers.  However, when you look at the RS485 tab on dashPanel what is your failure rate for inbound messages.

**Reply by jeefm** [reply]
100%

**Reply by rstrouse** [reply]
Tell me all about your setup.  Start with is this a pi3 or 4 or something else.  What RS485 adapter are u using?  Did you enable serial port in the os and disable the serial console.

**Reply by jeefm** [reply]
I'm using ELfin-EW11 Wifi connected to my intellitouch 4.  PoolController & dashpanel are running on a Windows machine.
**Config.json PoolController**
"controller": {
    "comms": {
      "portId": 0,
      "enabled": true,
      "rs485Port": "/dev/ttyUSB0",
      "mockPort": false,
      "netConnect": true,
      "netHost": "192.168.1.73",
      "netPort": 9899,
      "inactivityRetry": 10,
      "portSettings": {
        "baudRate": 9600,
        "dataBits": 8,
        "parity": "none",
        "stopBits": 1,
        "flowControl": false,
        "autoOpen": false,
        "lock": false

**Comment by jeefm** [comment]
sorry, on RS485 tab Send is 100% but receive is 0% but bytes keep growing.

**Reply by rstrouse** [reply]
Shoot a screenshot

**Reply by jeefm** [reply]
![rs485_stats](https://user-images.githubusercontent.com/88244843/189032659-02f9a3ca-cc07-44c4-a275-aa1070312521.png)

**Reply by tagyoureit** [reply]
Some more questions/ideas:
1. Do you have the Elfin setup with net connect/tcp server?
2. The default port on Elfin is 8899 but you have 9899.  Did you change it/does it match?
3. Elfin serial port settings should be 9600/8/1/n.
4. If all above are correct, did you try reversing the data wires?

**Reply by rstrouse** [reply]
On the Elfin configuration make sure the port is set to 9600 baud 8 data bits 1 stop bit and no parity.  If those are the settings try to flip the data wires.

**Reply by rstrouse** [reply]
Oops @tagyoureit beat me to it.

**Reply by jeefm** [reply]
![tcp](https://user-images.githubusercontent.com/88244843/189035133-a7bb5167-b61a-4674-9c4f-90735e5c0541.png)

**Comment by jeefm** [comment]
![netp](https://user-images.githubusercontent.com/88244843/189035168-a2cbdd7d-0c1b-4d37-bfc8-58f96de693cb.png)

**Comment by jeefm** [comment]
![tcp](https://user-images.githubusercontent.com/88244843/189035200-fba33fd2-f0a1-4ece-a487-b62b87fa0e5f.png)

**Comment by jeefm** [comment]
![serialport](https://user-images.githubusercontent.com/88244843/189035321-dd9aec86-549c-447b-9d34-940fd0f95e88.png)

**Reply by rstrouse** [reply]
The protocol should not be MQTT.  It should be over a tcp socket.

**Reply by jeefm** [reply]
I have a TCP server and MQTT, so I should get rid of MQTT tab?

**Reply by rstrouse** [reply]
It is not needed.  Not sure what the application for rs485 over MQTT would be.  What are the options under Cli settings.

**Comment by jeefm** [comment]
I tried to flip wires but I will try again.

**Reply by jeefm** [reply]
I flipped the wire.  I don't see collision anymore, but this error came up on the console of poolcontroller: Error sending setPumpRPMAsync.  Received Stats are now zero bytes.  I can assume the wiring was right previously?

**Reply by tagyoureit** [reply]
Something is still off. That error is when njsPC tries to control a pump (assuming you added one manually?). With Intellitouch we should never be controlling the pump directly but rather through your pentair OCP (outdoor control panel =Intellitouch). When the communications are working properly we will detect all of the equipment through the messages that the ocp is sending.

**Reply by jeefm** [reply]
![image](https://user-images.githubusercontent.com/88244843/189193445-5c1c444f-87d6-41be-9530-cea9a0989f6e.png)
Can this be the explanation?

**Reply by tagyoureit** [reply]
Nope.  The fact that you are seeing the connection closed and reopened every 10s means that we are not receiving any data.  The `Connection 0 connected` just means that we have opened the port properly.  

You probably saw the manual, but I'm attaching it here for convenience.  I would suggest that you look at page ~20 (5.2. TCP Server Test in AP Mode Step 1) and try to see if you can connect to the adapter.   
[elfin-ew1x-rs485-rs232-wifi-converter-manual.pdf](https://github.com/tagyoureit/nodejs-poolController/files/9529574/elfin-ew1x-rs485-rs232-wifi-converter-manual.pdf)

You were probably closer in your very first post where you were getting some bits but we couldn't understand them.  Now it seems you are not getting any bits or bytes coming through.
